csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/SourceGenerators/DotnetSourceGeneratorWrapper/DotnetSourceGeneratorWrapper.cs

```diff
@@ -33,11 +33,12 @@ public DotnetSourceGeneratorWrapper(
 
         protected abstract void GenerateAnalyzerConfig(IEnumerable<string> additionalFiles, string csprojFile, string analyzerConfigPath);
 
-        public IEnumerable<string> RunSourceGenerator(IEnumerable<string> additionalFiles, string csprojFile, IEnumerable<string> references, string targetDir)
+        public IEnumerable<string> RunSourceGenerator(IEnumerable<string> additionalFiles, string csprojFile, IEnumerable<string> references, string targetDir, string sourceDir)
         {
             try
             {
-                var name = Guid.NewGuid().ToString("N").ToUpper();
+                var relativePathToCsProj = Path.GetRelativePath(sourceDir, csprojFile);
+                var name = FileUtils.ComputeHash($"{relativePathToCsProj}\n{this.GetType().Name}");
```

Comment:
Is it correctly understood that it is now the callers responsibility to ensure that the csprojFile paths are unique (currently it is the case as this method is called on a range of dictionary keys)?

Reply:
Yes, the `csprojFile` should be unique, because we use that for grouping.